source/val/validation_state.cpp

```diff
@@ -425,6 +426,22 @@ bool ValidationState_t::RegisterUniqueTypeDeclaration(
     key.insert(key.end(), inst.words + words_begin, inst.words + words_end);
   }
 
+  if (inst.opcode == SpvOpTypePointer &&
+      HasExtension(Extension::kSPV_KHR_variable_pointers)) {
+    const std::vector<Decoration>& decorations = id_decorations(inst.result_id);
+    std::map<SpvDecoration, const std::vector<uint32_t>*> decoration_params;
+    for (const auto& decoration : decorations) {
+      const auto result = decoration_params.emplace(decoration.dec_type(),
+                                                    &decoration.params());
+      (void)result;
+      assert(result.second);
```

Comment:
I don't think you should assert here? There's nothing wrong with redundant decorations, except that it wastes space.
I don't think the storage behind id_decorations has unique'd it.

Reply:
It's worth saying you're assuming that a single decoration enum can only be set once per target ID.
It think that's true currently, and is likely to be true always.  But I don't recall a SPIR-V validation rule about that. (Perhaps an oversight.)

Reply:
I will assume that decorations are not unique, but only use the last one.